src/Bearded.TD/Game/Simulation/Elements/ElementalPhenomenonScopeBase.cs

```diff
@@ -28,57 +32,48 @@ public void Adopt(TEffect effect, Instant now)
 
     public void ApplyTick(Instant now)
     {
-        activeEffects.RemoveAll(e => e.Expiry <= now);
+        activeEffects.RemoveAll(now, static (t, e) => e.Expiry <= t);
 
-        if (TryChooseEffect(out var effect))
+        if (TryChooseEffect(CollectionsMarshal.AsSpan(activeEffects)) is { } effect)
         {
             transitionToEffect(effect);
-            ApplyEffectTick(target, effect);
+            ApplyEffectTick(effect);
         }
         else
         {
             endEffectIfPreviouslyActive();
         }
 
-        if (activeEffect?.StatusIcon is not null)
+        if (activeEffect?.StatusIcon is { } icon)
         {
-            updateStatusIconExpiry(activeEffect.StatusIcon);
+            updateStatusIconExpiry(icon);
         }
     }
 
     private void transitionToEffect(TEffect effect)
     {
         // null -> effect A
-        if (activeEffect is null)
+        if (activeEffect is not { } current)
         {
-            StartScope(target, out var createStatus);
-            startEffect(effect, out var createOverrideStatus);
-            createStatus = createOverrideStatus ?? createStatus;
-            var receipt = createStatus is null ? null : reportStatus(createStatus);
+            EffectChangeResult? statusChange = null;
```

Comment:
Again, maybe I am misunderstanding how nullable works in C# these days, but a nullable struct is essentially an allocated class anyway, right? Do we still need the `ref` in this case?